### DISCUSSION TITLE
Can I temporarily suppress delta?

### DISCUSSION THREAD

**Discussion by borekb** [dandavison/delta#735] [Q&A] [answered]
If I have delta in my `.gitconfig`, is there a way to quickly disable it for one specific `git ...` invocation? Something like `DISABLE_DELTA=1 git diff ...` or similar.

**Comment by dandavison** [comment, marked as answer]
Hi @borekb, yes, for example:

```
git -c pager.diff='less -R' diff
```

or `pager.show` etc. There's also `core.pager`, but `pager.diff`  / `pager.show` etc is what delta's README recommends. And there's also `git -P` ("no pager").

**Reply by borekb** [reply]
Thanks. So if I understand correctly, I need to do the right thing depending on the git command but there's no way to use my "usual defaults" (as if I commented out a delta section in my `.gitconfig`), right? Just want to make sure, maybe it's not technically possible to disable / suppress delta the way I imagine...

**Reply by dandavison** [reply]
I agree that superficially it's a bit inconvenient. However, the following methods do not depend on the current command

1. You can use `git -P ...` if it is acceptable not to use a pager at all
2. You could create a git alias, or a shell alias, along the following lines ("nuclear option"): `git -c core.pager=$PAGER -c pager.diff=$PAGER -c pager.show=$PAGER -c pager.log=$PAGER ...` That's assuming $PAGER holds something like `less -R`, but you can hardcode that explicitly of course.

Delta can't really make this more convenient -- it's about how we tell git not to use its configured pager.

**Reply by borekb** [reply]
Great, I'll probably use `-P`, it's fine for the rare few occasions where I don't want to use delta 😄. Thanks again!

**Reply by dandavison** [reply]
No problem!

**Reply by fakhrulhilal** [reply]
I used to use `git --no-pager diff > file.patch` to create a patch. The same way as option 1, just using long parameter style. Remember to use `--no-pager` or `-P` before other git command.

**Comment by mcornella** [comment]
`GIT_PAGER=` also works, for example if you are using an alias it might be faster:

```sh
GIT_PAGER= gd # alias to git diff
```
